packages/aws-cdk-lib/aws-codepipeline/lib/pipeline.ts

```diff
@@ -325,7 +332,9 @@ export class Pipeline extends PipelineBase {
       public readonly pipelineArn = pipelineArn;
     }
 
-    return new Import(scope, id);
+    return new Import(scope, id, {
+      environmentFromArn: pipelineArn,
+    });
```

Comment:
This is just a bug fix to codepipeline that we should have had in a long time ago lol